tools/isolated-tests/src/RunTestsIsolatedCommand.php

```diff
@@ -0,0 +1,172 @@
+    }
+
+    protected function configure(): void
+    {
+        $this->setDescription('Runs every test isolated from the other.');
+        $this->addOption('depth', null, InputOption::VALUE_REQUIRED, '1 for test classes, 2 for test methods, 3 for every single provider data set', '3');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        if (false === $this->phpPath) {
+            throw new \RuntimeException('The php executable could not be found.');
+        }
+
+        $depth = (int) $input->getOption('depth');
+
+        if ($depth < 1 || $depth > 3) {
+            throw new \InvalidArgumentException('--depth must be an integer between 1 and 3');
+        }
+
+        $php = [
+            $this->phpPath,
+            '-dmemory_limit='.ini_get('memory_limit'),
+        ];
+
+        if (OutputInterface::VERBOSITY_DEBUG === $output->getVerbosity()) {
+            $php[] = '-ddisplay_errors=-1';
+            $php[] = '-ddisplay_startup_errors=-1';
+        }
+
+        $phpunit = $this->projectDir.'/vendor/bin/phpunit';
+
+        $listOutput = new BufferedOutput();
+
+        $this->executeCommand(array_merge($php, [$phpunit, '--list-tests']), $listOutput);
+
+        $tests = [[], [], []];
+
+        foreach (preg_split('/\r?\n/', $listOutput->fetch()) as $line) {
+            if (preg_match('/^ - (\S+)(::[^\s#"]+)(.*)$/', $line, $matches)) {
+                $tests[0][] = $matches[1];
+
+                if ($depth > 1) {
+                    $tests[1][] = $matches[1].$matches[2];
+                }
+
+                if ($matches[3] && $depth > 2) {
+                    $tests[2][] .= $matches[1].$matches[2].' with data set '.$matches[3];
+                }
+            }
+        }
+
+        $tests = array_values(array_unique(array_merge(...$tests)));
+
+        $commandFlags = array_filter([
+            $output->isDecorated() ? '--colors=always' : '',
+            $this->getVerbosityFlag($output),
+        ]);
+
+        $failedTests = [];
+
+        foreach ($tests as $test) {
+            // Skip if the whole class, or the test with all data sets failed already
+            foreach ($failedTests as $failedTest) {
+                if (0 === strncmp($test, $failedTest, \strlen($failedTest))) {
+                    continue 2;
+                }
+            }
+
+            $filter = preg_quote($test);
+            $output->writeln("> vendor/bin/phpunit --filter '".str_replace("'", '\'"\'"\'', $filter)."'");
+
+            try {
+                $this->executeCommand(array_merge($php, [$phpunit, '--extensions', GlobalStateWatcher::class, '--filter', $filter], $commandFlags), $buffer = new BufferedOutput());
+
+                // Clear previously written line
+                $output->write("\e[1A\e[K");
+            } catch (\Throwable $e) {
+                $failedTests[] = $test;
+                $output->writeln($buffer->fetch());
```

Comment:
Do we need to handle this?

<img width="706" alt="" src="https://user-images.githubusercontent.com/1192057/154008436-1744e046-d79c-4f43-b47c-2d22b5d98408.png">

Reply:
👍
Fixed in 8b9242045b9fdf7e0bc4f3f347b4d44412eda5a1